test/jdk/java/awt/Focus/AppletInitialFocusTest/AppletInitialFocusTest1.java

```diff
@@ -21,65 +21,51 @@
  * questions.
  */
 
-import java.awt.*;
-import java.awt.event.*;
+import java.awt.Button;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
 
-/*
-  @test
-  @key headful
-  @bug 4411534 4517274
-  @summary ensures that user's requestFocus() during applet initialization
-           is not ignored
-  @library ../../regtesthelpers
-  @build   Util
-  @run main AppletInitialFocusTest1
+/**
+ * @test
+ * @key headful
+ * @bug 4411534 4517274
+ * @summary ensures that user's requestFocus() during applet initialization
+ *          is not ignored
  */
 public class AppletInitialFocusTest1 extends Frame implements FocusListener {
 
     Button button1 = new Button("Button1");
     Button button2 = new Button("Button2");
-
-    Object lock = new Object();
+    private static volatile Object focused;
 
     public static void main(final String[] args) throws Exception {
         AppletInitialFocusTest1 app = new AppletInitialFocusTest1();
-        app.init();
-        Thread.sleep(10000);
-    }
-
-    public void init() {
-        setSize(200, 200);
-        setLocationRelativeTo(null);
-        setLayout(new FlowLayout());
+        try {
+            app.setSize(200, 200);
+            app.setLocationRelativeTo(null);
+            app.setLayout(new FlowLayout());
 
-        Component parent = this;
-        while (parent != null && !(parent instanceof Window)) {
-            parent = parent.getParent();
-        }
-        /*
-         * This applet is designed to be run only with appletviewer,
```

Comment:
This is the old code assumed that the test is executed via appletviewer as an applet.